archunit/src/main/java/com/tngtech/archunit/library/Architectures.java

```diff
@@ -289,4 +292,111 @@ public String toString() {
             }
         }
     }
+
+    @PublicAPI(usage = ACCESS)
+    public static OnionArchitecture onionArchitecture() {
+        return new OnionArchitecture();
+    }
+
+    public static final class OnionArchitecture implements ArchRule {
+        private static final String DOMAIN_MODEL_LAYER = "domain model";
+        private static final String DOMAIN_SERVICE_LAYER = "domain service";
+        private static final String APPLICATION_LAYER = "application";
+        private static final String ADAPTER_LAYER = "adapter";
+
+        private String[] domainModelPackageIdentifiers = new String[0];
+        private String[] domainServicePackageIdentifiers = new String[0];
+        private String[] applicationPackageIdentifiers = new String[0];
+        private Map<String, String[]> adapterPackageIdentifiers = new LinkedHashMap<>();
+
+        private LayeredArchitecture layeredArchitectureDelegate = layeredArchitecture();
```

Comment:
I don't think I would even keep a field here. Make `updateLayersDelegate()` a `layeredArchitectureDelegate()` returning a new `layeredArchitecture()` with the current configuration every time. Yes, you might create it more often than necessary, but I'm fairly sure that's not where the performance problems are :wink: 
Especially because you want to be able to query the `getDescription()` at every point in time, and now it only works after `evaluate` has been called. I think the code is easier without the field and keeping anything in sync, simply create the layered architecture whenever needed. Similar if you use the `as(..)` method.
Maybe a unit test for these trivial things like `as(..)` and `because(..)` wouldn't be so bad :wink: 
You might have to keep a separate field `description` though, if you want a description independent of `LayeredArchitecture's` description.

Reply:
Thanks for the hint. Sounds very reasonable and I changed the code accordingly. In particular, I now understand `as` and `because` ;)

Reply:
I also changed the `description` accordingly.

Reply:
Unfortunately, I still cannot build the `docs`. Am I blind or what is wrong with my setup? I already tried building the `docs` folder with `./gradlew build` but always get the following error:

```text
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Project directory '<SOME_LOCAL_PATH>/TNG/ArchUnit/docs' is not part of the build defined by settings file '<SOME_LOCAL_PATH>/TNG/ArchUnit/settings.gradle'. If this is an unrelated build, it must have its own settings file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
```